### PR TITLE
Continuing work on homepage

### PR DESCRIPTION
**Pull request by C4L86**

### CODE REVIEW

File: app/models/genre.rb [hunk 1]
```diff
@@ -1,3 +1,8 @@
 class Genre < ActiveRecord::Base
 has_many :albums
+
+  # Return a count of albums associated with a certain genre
+  def album_count
```
Comment (sumeetjain) [inline app/models/genre.rb:5]: Inconsistent indentation detected.

**Comment by C4L86** [comment]
I'm attempting to define some methods to return the count of artists associated with a given area, albums with a given genre, and release with a given format. At the moment, I am getting returns of ActiveRecord relations and am not sure how to proceed from here. My brain has apparently gone to mush since putting this kind of work on the back burner for a while.

**Comment by sumeetjain** [comment]
ActiveRecord Relations are a lot like Arrays. If you had this array:

``` ruby
artists = [
  Artist1,
  Artist2,
  Artist3
]
```

and if each artist had a method called `number_of_members`, which returned an integer with the number of members in that band, then...

``` ruby
sum = 0

artists.each do |a|
  sum += artist.number_of_members
end

return sum
```

would get you a sum.

This is pseudo-code, sort of. But let me know if it doesn't help you move forward.

**Comment by C4L86** [comment]
I worked around the probelm in a way that I am guessing isn't best practice. In the albums index View, I have:

**Comment by C4L86** [comment]
I worked around the problem in a way that I think is probably not best practice as the logic should be contained in the Model when possible. In the albums index View, I have:

```
<% @areas.each do |area| %>
  <br />
  <%= area.name %>
  (<%= Artist.where({"area_id" => area.id}).count %>)
<% end %>
```

It's the same for the other instances in which I was trying to evoke the counts.
Before asking for assistance, I had attempted to define the methods in their respective Models like this:

```
class Area
  def artist_count
    Artist.where({"area_id" => area.id}).count
  end
end
```

but would receive an error in doing so. I am trying to understand why this code works when not being used in a method but creates an error when used like in the method defined above. Could you provide some insight?

**Comment by sumeetjain** [comment]
Within the `artist_count` method, `area` is undefined.

Remember that when you're creating methods in a class, you're defining how the class will behave. The methods you write are entirely abstract–they don't have access to other local variables (like `area`) that might get defined when that class's objects are actually used.

This is what `self` is for. It's a reference to the object itself, in the moment when it's being called.

Changing your method to:

``` ruby
Artist.where({"area_id" => self.id}).count
```

would probably fix things for you.

**Comment by C4L86** [comment]
I'm sorry, I had defined the methods with `self.id`, but would receive the following error:
screen shot 2016-03-27 at 11 49 45 am

I suppose this is why I am confused as to why the line of code works outside of the method but not within.

Side note:
Was I correct in assuming that the line of code contained in the method shouldn't be in the View?

**Comment by sumeetjain** [comment]
Based on the screenshot, it looks like you typed `area.id` instead of `area_id`.

> Was I correct in assuming that the line of code contained in the method shouldn't be in the View?

Yes. While it should work either way, the view should contain as little logic as possible.

**Comment by C4L86** [comment]
_forehead slap forever_ Thank you for taking the time out of your day for such a silly mistake. I knew that it should've been working, sans the typo. Problem solved.

**Pull request merged**